lightwood/api/json_ai.py

```diff
@@ -185,15 +185,18 @@ def generate_json_ai(
     exec(IMPORT_EXTERNAL_DIRS, globals())
     target = problem_definition.target
     input_cols = []
+    tss = problem_definition.timeseries_settings
+
     for col_name, col_dtype in type_information.dtypes.items():
         if (
-            col_name not in type_information.identifiers
-            and col_dtype not in (dtype.invalid, dtype.empty)
-            and col_name != target
+                (col_name not in type_information.identifiers
+                 and col_dtype not in (dtype.invalid, dtype.empty)
+                 and col_name != target)
+                or
+                (tss.group_by is not None and col_name in tss.group_by)
```

Comment:
how did this get missed before? Isn't it by definition not an : [identifier, invalid, empty, target] type?

Reply:
Nope, it will get tagged as `invalid` (if I recall correctly) when there's a single value in the column.